src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -122,6 +132,189 @@ private PackageArchiveReader BuildFromProjectJson(string path)
             return package;
         }
 
+        private void BuildProjectWithDotNet()
+        {
+            string properties = string.Empty;
+            if (_packArgs.Properties.Any())
+            {
+                foreach (var property in _packArgs.Properties)
+                {
+                    if (property.Key.Equals(Configuration, StringComparison.OrdinalIgnoreCase))
+                    {
+                        properties = $"-c {property.Value}";
+                    }
+                }
+            }
+
+            if (!string.IsNullOrEmpty(_packArgs.OutputDirectory))
+            {
+                string outputDirectory = _packArgs.OutputDirectory;
+                properties += $" -b \"{outputDirectory}\"";
+            }
+
+            if (_dotnetLocation == null)
+            {
+                GetDotNetLocation();
+            }
+
+            var processStartInfo = new ProcessStartInfo
+            {
+                UseShellExecute = false,
+                FileName = _dotnetLocation,
+                Arguments = $"build {properties}",
+                WorkingDirectory = _packArgs.BasePath,
+                RedirectStandardOutput = false,
+                RedirectStandardError = false
+            };
+
+            int result;
+            using (var process = Process.Start(processStartInfo))
+            {
+                process.WaitForExit();
+
+                result = process.ExitCode;
+            }
+
+            if (0 != result)
+            {
+                // If the build fails, report the error
+                throw new Exception(String.Format(CultureInfo.CurrentCulture, Strings.Error_BuildFailed, processStartInfo.FileName, processStartInfo.Arguments));
+            }
+        }
+
+        private void GetDotNetLocation()
+        {
```

Comment:
This looks like it should go into ``NuGetEnvironment`` and also use some of the paths contained there.